spec/forms/waste_exemptions_engine/check_contact_name_form_spec.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+module WasteExemptionsEngine
+  RSpec.describe CheckContactNameForm, type: :model do
+    it_behaves_like "a validated form", :check_contact_name_form do
+      let(:valid_params) do
+        [
+          { temp_reuse_applicant_name: "true" },
+          { temp_reuse_applicant_name: "false" }
+        ]
+      end
+      let(:invalid_params) do
+        [
+          { temp_reuse_applicant_name: "" }
+        ]
+      end
+    end
+
+    describe "#submit" do
+      let(:form) { build(:check_contact_name_form) }
+
+      subject do
+        form.submit(temp_reuse_applicant_name: temp_reuse_applicant_name)
+      end
+
+      context "when temp_reuse_applicant_name is true" do
+        let(:temp_reuse_applicant_name) { "true" }
+
+        it "assigns the applicant name as the contact name" do
+          subject
+
+          expect(form.contact_first_name).to eq(form.applicant_first_name)
+          expect(form.contact_last_name).to eq(form.applicant_last_name)
+        end
+      end
+
+      context "when temp_reuse_applicant_name is false" do
```

Comment:
does this need to include a spec for if `temp_reuse_applicant_name` is empty/blank?

Reply:
This is handled by `validates :temp_reuse_applicant_name, "defra_ruby/validators/true_false": true` in `app/forms/waste_exemptions_engine/check_contact_name_form.rb` - I agree it could do with a spec.